api/test/test_helper/android_plugin_mock.gd

```diff
@@ -0,0 +1,22 @@
+# ##############################################################################
+#
+#	Copyright 2020 Google LLC
+#
+#	Licensed under the Apache License, Version 2.0 (the "License");
+#	you may not use this file except in compliance with the License.
+#	You may obtain a copy of the License at
+#
+#		https://www.apache.org/licenses/LICENSE-2.0
+#
+#	Unless required by applicable law or agreed to in writing, software
+#	distributed under the License is distributed on an "AS IS" BASIS,
+#	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#	See the License for the specific language governing permissions and
+#	limitations under the License.
+#
+# ##############################################################################
+class_name AndroidPluginMock
```

Comment:
Rather than making this a blank class and mocking out the implementation in the tests, consider making this a `FakeAndroidPlugin` that roughly simulates the real plugin. You could add a `addDownloadedPack(AssetPackLocation)` method that allows you to specify the result of getPackLocation() without mocking.

Later on, you could implement `fetch` on the fake plugin to produce a specified series of stateUpdated signals that you could use to test.

Reply:
Added a FakeAndroidPlugin class and removed the empty AndroidPluginMock class. 
I think this design will be especially useful when we start testing the asynchronous signals, since Gut only has limited ability on mocking side effects. With this approach we would have more granular control over plugin simulation/mocking.